packages/ckeditor5-ui/src/editorui/poweredby.ts

```diff
@@ -352,14 +352,21 @@ function getLowerCornerPosition(
 
 			if ( firstScrollableEditableElementAncestor ) {
 				const firstScrollableEditableElementAncestorRect = new Rect( firstScrollableEditableElementAncestor );
+				const notVisibleVertically = visibleEditableElementRect.bottom + balloonRect.height / 2 >
+				firstScrollableEditableElementAncestorRect.bottom;
+				const notVisibleHorizontally = editableElementRect.right > firstScrollableEditableElementAncestorRect.right;
 
-				// The watermark cannot be positioned in this corner because the corner is "not visible enough".
-				if ( visibleEditableElementRect.bottom + balloonRect.height / 2 > firstScrollableEditableElementAncestorRect.bottom ) {
+				// The watermark cannot be positioned in this corner because the corner is "not visible enough" vertically.
+				if ( notVisibleVertically ) {
+					return OFF_THE_SCREEN_POSITION;
+				}
+
+				// The watermark cannot be positioned in this corner because the corner is "not visible enough" horizontally.
+				if ( notVisibleHorizontally ) {
```

Comment:
I truly don't like this kind of condition, because that way for 100% coverage you need only one test, indeed we have here two reasons for `OFF_THE_SCREEN_POSITION` , so this style of code demands writing 2 tests for 100% coverage. After adding few lines of code in this PR  coverage was still 100%, but we have new logic, so to my mind it should not be like that, so that's why I created new condition like this.

Reply:
1.  Imagine that you've got more than two conditions to check that returns same value.
2.  Code readability - its easier to take a look at one \`if\` than on few that returns always the same value.
3.  100% CC - you need to write those tests in way that they need to be functional and logical that covers a real testing cases, not just to make blindly 100% and thats it (workshop about writing tests - we must finally do it :smile: ).